pkg/prowgen/prowgen.go

```diff
@@ -333,18 +333,39 @@ func generatePodSpecMultiStage(info *ProwgenInfo, test *cioperatorapi.TestStepCo
 		secrets = append(secrets, &api.Secret{Name: api.HiveControlPlaneKubeconfigSecret})
 	}
 	podSpec := generateCiOperatorPodSpec(info, secrets, []string{test.As}, additionalArgs...)
-	if profile == "" {
-		return podSpec
+
+	if profile != "" {
+		podSpec.Volumes = append(podSpec.Volumes, generateClusterProfileVolume(profile, profile.ClusterType()))
+		clusterProfilePath := fmt.Sprintf("/usr/local/%s-cluster-profile", test.As)
+		container := &podSpec.Containers[0]
+		container.Args = append(container.Args, fmt.Sprintf("--secret-dir=%s", clusterProfilePath))
+		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{Name: "cluster-profile", MountPath: clusterProfilePath})
 	}
-	podSpec.Volumes = append(podSpec.Volumes, generateClusterProfileVolume(profile, profile.ClusterType()))
-	clusterProfilePath := fmt.Sprintf("/usr/local/%s-cluster-profile", test.As)
-	container := &podSpec.Containers[0]
-	container.Args = append(container.Args, fmt.Sprintf("--secret-dir=%s", clusterProfilePath))
-	container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{Name: "cluster-profile", MountPath: clusterProfilePath})
-	addLeaseClient(podSpec)
+
+	if testContainsLease(test) || profile != "" {
+		addLeaseClient(podSpec)
+	}
+
 	return podSpec
 }
 
+func testContainsLease(test *cioperatorapi.TestStepConfiguration) bool {
+	// this is predicated upon the config being fully resolved at this time.
+	if test.MultiStageTestConfigurationLiteral == nil {
+		return false
+	}
+
+	hasLeases := false
+	for _, test := range test.MultiStageTestConfigurationLiteral.Test {
```

Comment:
This also needs to look at leases declared in steps, see for example:

https://github.com/openshift/ci-tools/blob/0521ae628982b19092d6d38bb9cb92d945474cc1/pkg/defaults/defaults.go#L484-L500

Reply:
We could also take the opportunity to create some proper functions in the `api` package for these now that they're needed elsewhere.